crates/ruff/src/linter.rs

```diff
@@ -260,7 +264,7 @@ pub fn add_noqa_to_path(path: &Path, package: Option<&Path>, settings: &Settings
     let contents = std::fs::read_to_string(path)?;
 
     // Tokenize once.
-    let tokens: Vec<LexResult> = ruff_python_parser::tokenize(&contents);
+    let tokens: Vec<LexResult> = ruff_python_parser::tokenize(&contents, Mode::Module);
```

Comment:
How do we ensure that we pass the correct `mode` in all the places where we call `tokenize`, `lex` or `lex_starts_at` today? 

I haven't thought this through fully but I start wondering if we should have something like a `PythonFile` that stores the source, mode, required python version, etc. and provides method for parsing and lexing the content.